### ISSUE TITLE
GitFetchStrategy has no direct support for required submodules

### ISSUE THREAD

**Issue by trws** [LLNL/spack#983]
This is related to the construction of the rust package in #982, there are no options for handling submodules in the git fetcher at the moment.  I went to create this, but there are enough options for how to handle it as to convince me to start a discussion first.  Options:
1.  Just add `--recursive` to clone when requested
   - This is the simplest, longest supported option, but it means cloning the entirety of all submodules, which is painful when one of them is LLVM or similar
2.  Add an option that causes `git submodule update --init --recursive` to be run after clone
   - Basically equivalent to the first option, but could include `--depth` if it's a new enough git, but `--depth` may fail on some versions and repositories if the commit in question is not within that depth
3. Something else:
   - There could always be per-directory submodule handling, a separate command... whatever, any thoughts?

**Comment by RalphTheBest** [comment]
Hi,

I'm new to spack and the idea may be dumb, but this feature seems interesting for my work, so here goes nothing. 
Could it be reasonable to try the second option (with `--depth` to make lighter the process) and if it fails (repo or version too old to support it, or any other reason) use the first option. (Heavier, but granting the success).

Would it be possible/correct? IMHO, that could be a good way to go.
